pcsx2/Gif.cpp

```diff
@@ -194,12 +311,15 @@ static __fi tDMA_TAG* ReadTag2()
 
 bool CheckPaths(EE_EventType Channel) {
 	// Can't do Path 3, so try dma again later...
-	if(!gifUnit.CanDoPath3()) {
-		if(!gifUnit.Path3Masked())
-		{
-			CPU_INT(Channel, 128);
+	if (!CHECK_GIFFIFOHACK) {
+		if (!gifUnit.CanDoPath3()) {
+			if (!gifUnit.Path3Masked())
+			{
```

Comment:
Because if gifUnit.CanDoPath3() is false  and gifUnit.Path3Masked()  is false, we need to tell the GIF to loop to keep checking for when it can continue, if you just return true it will continue and it shouldn't be doing that.